commit 659634effe22
Author: Valentin Gosu <valentin.gosu@gmail.com>
Date:   Wed Apr 4 13:47:46 2018 +0300

    Bug 1451262 - Clear gSocketThread after finishing all operations so we don't trigger thread assertions r=mayhemer
    
    MozReview-Commit-ID: AHMPd72S2GB
    
    --HG--
    extra : rebase_source : fe5fb1604e771f223d78f23fcf9d37b1f85763f9
---
 netwerk/base/nsSocketTransportService2.cpp | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git netwerk/base/nsSocketTransportService2.cpp netwerk/base/nsSocketTransportService2.cpp
index abaa8f8fb6da3..a16d888998ae9 100644
--- netwerk/base/nsSocketTransportService2.cpp
+++ netwerk/base/nsSocketTransportService2.cpp
@@ -1067,9 +1067,8 @@ nsSocketTransportService::Run()
     // socket detach handlers get processed.
     NS_ProcessPendingEvents(mRawThread);
 
-    gSocketThread = nullptr;
-
     psm::StopSSLServerCertVerificationThreads();
+    gSocketThread = nullptr;
 
     SOCKET_LOG(("STS thread exit\n"));
 
commit 43e3838b681b
Author: Valentin Gosu <valentin.gosu@gmail.com>
Date:   Thu May 10 10:37:24 2018 +0200

    Bug 1451262 - Call NS_ProcessPendingEvents after StopSSLServerCertVerificationThreads. r=mayhemer, a=RyanVM
    
    If there are still pending events after we null out gSocketThread, OnSocketThread() will always return false, which can trigger assertions.
    
    MozReview-Commit-ID: CovFIU3cKL
    
    --HG--
    extra : source : b859a16d74500eed3b00d7e0bc1eba0d9e0c34a4
---
 netwerk/base/nsSocketTransportService2.cpp | 5 +++++
 1 file changed, 5 insertions(+)

diff --git netwerk/base/nsSocketTransportService2.cpp netwerk/base/nsSocketTransportService2.cpp
index 637a7d95f748c..06cda53f52056 100644
--- netwerk/base/nsSocketTransportService2.cpp
+++ netwerk/base/nsSocketTransportService2.cpp
@@ -1067,7 +1067,12 @@ nsSocketTransportService::Run()
     // socket detach handlers get processed.
     NS_ProcessPendingEvents(mRawThread);
 
+    // Stopping the SLL threads can generate new events, so we need to
+    // process them before nulling out gSocketThread, otherwise we can get
+    // !onSocketThread assertions.
     psm::StopSSLServerCertVerificationThreads();
+    NS_ProcessPendingEvents(mRawThread);
+
     gSocketThread = nullptr;
 
     SOCKET_LOG(("STS thread exit\n"));
